Adding Support for interlude audio Hats.

Yash-BlueRockSoft opened the pull request:
Interlude Audio has created 2 hats for the raspberry to be put up for sale online at www.interludeaudio.com the hats are based of the cs4272 and wm8805 codec and spdif chip which which leverage pre-existing drivers, enhanced with support for the unique features of our devices. We hope to mainline our changes to the kernel so as to have them readily available for inclusion in future releases.

pelwell commented on sound/soc/bcm/rpi-wm8804-soundcard.c:
```diff
@@ -372,6 +474,20 @@ static int snd_rpi_wm8804_probe(struct platform_device *pdev)
 		snd_clk48gpio =
 			devm_gpiod_get(&pdev->dev, "clock48", GPIOD_OUT_LOW);
 
+		led_gpio_1 = devm_gpiod_get(&pdev->dev, "led1", GPIOD_OUT_LOW);
```
Board-specific code like this needs to go into the board-specific probe function, in this case snd_interlude_audio_digital_probe.

pelwell commented on sound/soc/bcm/rpi-wm8804-soundcard.c:
```diff
@@ -140,26 +174,45 @@ static int snd_rpi_wm8804_hw_params(struct snd_pcm_substream *substream,
 		dev_err(rtd->card->dev,
 		"Failed to set WM8804 SYSCLK, unsupported samplerate %d\n",
 		samplerate);
+
```
Don't make random whitespace changes.

pelwell commented on sound/soc/bcm/rpi-wm8804-soundcard.c:
```diff
@@ -127,7 +162,6 @@ static int snd_rpi_wm8804_hw_params(struct snd_pcm_substream *substream,
 
 	/* clear until all clocks are setup properly */
 	wm8804_samplerate = 0;
-
```
Don't make random whitespace changes.

pelwell commented on sound/soc/bcm/rpi-wm8804-soundcard.c:
```diff
@@ -140,26 +174,45 @@ static int snd_rpi_wm8804_hw_params(struct snd_pcm_substream *substream,
 		break;
 	case 44100:
 		sampling_freq = 0x00;
+		gpiod_set_value_cansleep(led_gpio_1, 1);
```
This code is shared with all the other wm8804-based cards. You should create your own hw_params function that calls this one and also sets the LEDs as you want them. You can hook that in by providing a .ops member to snd_interlude_audio_digital_dai.

pelwell commented on sound/soc/bcm/rpi-wm8804-soundcard.c:
```diff
@@ -372,6 +474,20 @@ static int snd_rpi_wm8804_probe(struct platform_device *pdev)
+		mdelay(10);
+		gpiod_set_value_cansleep(custom_reset, 1);
+		if(of_property_read_u32(pdev->dev.of_node, "sys_clk", &sysclk_freq))
+		{
```
This warning (error!) is unnecessary. Delete it and info message.

pelwell reviewed (changes requested):
It's good that you are adding your code to an existing driver, but it shouldn't impact other users of the driver. Go through the review comments and checkpatch complaints, squash the results down to two commits - a driver patch and one for the overlays - then force push it so we can review the result.

benpayne committed:
Impliment driver support for Interlude Audio Digital Hat

Implementing driver support for
Interlude audio's WM8805 based digital hat
by leveraging existing drivers

benpayne committed:
Add overlays needed for Interlude Audio Digital and Analog hats

Adding 2 new overlays for use with
Interlude Audio's Digital and Analog hats
adding descriptions for both in README
adding changes to Makefile to include both DT's

Yash-BlueRockSoft commented:
Hello @pelwell all requested changes have been completed we have squashed the changes down to 2 commits and have tested the modified driver code and confirmed it works; look forward to your review

pelwell commented on sound/soc/bcm/rpi-wm8804-soundcard.c:
```diff
@@ -287,6 +371,54 @@ static struct snd_rpi_wm8804_drvdata drvdata_hifiberry_digi = {
+
+static int snd_interlude_audio_digital_probe(struct platform_device *pdev)
+{
+	pr_debug("%s\n", __func__);
```
You can drop this line now.

pelwell commented on sound/soc/bcm/rpi-wm8804-soundcard.c:
```diff
@@ -65,15 +66,52 @@ struct snd_rpi_wm8804_drvdata {
+{
+	snd_soc_add_component_controls(component, wm8805_input_controls_card,
+	ARRAY_SIZE(wm8805_input_controls_card));
+	pr_debug("adding new controls");
```
Drop this line.

pelwell commented on sound/soc/bcm/rpi-wm8804-soundcard.c:
```diff
@@ -372,6 +506,16 @@ static int snd_rpi_wm8804_probe(struct platform_device *pdev)
 		snd_clk48gpio =
 			devm_gpiod_get(&pdev->dev, "clock48", GPIOD_OUT_LOW);
 
+		custom_reset = devm_gpiod_get(&pdev->dev, "reset", GPIOD_OUT_LOW);
```
All of this can go in your custom probe function.

pelwell reviewed (changes requested):
I think you're nearly there, but there are a few improvements still to be made. See the inline comments. Also, please fix the checkpatch errors and warnings: https://github.com/raspberrypi/linux/actions/runs/7908859459/job/21601350819?pr=5949#step:6:74

pelwell commented on sound/soc/bcm/rpi-wm8804-soundcard.c:
```diff
@@ -372,6 +512,12 @@ static int snd_rpi_wm8804_probe(struct platform_device *pdev)
 		snd_clk48gpio =
 			devm_gpiod_get(&pdev->dev, "clock48", GPIOD_OUT_LOW);
 
+		if(of_property_read_u32(pdev->dev.of_node, "sys_clk", &sysclk_freq))
```
I meant all of it, including the sys_clk logic - you have no reason to make changes for the other drivers, particularly the kernel log spam.

Yash-BlueRockSoft replied:
okay sounds good changes have been made

Yash-BlueRockSoft commented:
All changes have been made the build tests and checkpatch review action should pass as well

pelwell commented:
If you're happy with that then so am I. I'll merge once the build checks have completed.

Yash-BlueRockSoft commented:
Thanks for your help
